src/containers/Routing/Routing.jsx

```diff
@@ -15,9 +15,17 @@ import Splash from "../Splash/Splash";
 import CreateAccount from "../CreateAccount/CreateAccount";
 import Articles from "../Articles/Articles";
 import SkillsTree from "../SkillsTree/SkillsTree";
+import { auth } from "../../firebase";
 import Challenge from "../Challenge/Challenge";
 
 const Routing = () => {
+
+  const [user, setUser] = useState({ firstName: "", lastName: "", email: "", password: "" });
+  
+  const handleInputChange = (event) => {
+    setUser(previousState => ({ ...previousState, [event.target.name]: event.target.value }));
+  }
+
```

Comment:
At the moment if you refresh you lose your user. Firebase gives us a way to handle this [onAuthStateChanged](https://firebase.google.com/docs/auth/web/manage-users#get_the_currently_signed-in_user)

You should be able to put this in a `useEffect` which should be able to catch any changes.

```js
  useEffect(() => {
    onAuthStateChanged(auth, (authenticatedUser) => {
      if (authenticatedUser) {
        setUser(authenticatedUser);
      } else {
        setUser(null);
      }
    });
  }, []);
```